src/parameterizations/lateral/MOM_thickness_diffuse.F90

```diff
@@ -333,6 +335,19 @@ subroutine thickness_diffuse(h, uhtr, vhtr, tv, dt, G, GV, US, MEKE, VarMix, CDp
     endif
   endif
 
+  if (use_VarMix) then
+    if (use_gradient_model) then  !< Gradient model (Khani & Dawson, JAMES 2023)
+      if (CS%grad_Khani_scale > 0.0) then
+      !$OMP do
+        do k=1,nz ; do j=js,je ; do I=is-1,ie
+          KH_u(I,j,k) = 1.0 * CS%grad_Khani_scale * VarMix%L2grad_u(I,j) * VarMix%UH_grad(I,j,k)
```

Comment:
As the code is currently written, this expression is dimensionally inconsistent, especially in non-Boussinesq mode.  The left hand side has units of `[L2 T-1 ~> m2 s-1]`, whereas the right hand side has units of `[L2 ~> m2] * [H L-1 T-1 ~> s-1 or kg m-3 s-1]` which combines into `[H L T-1 ~> m2 s-1 or kg m-1 s-1]`, although I suspect that this will be sorted out when the dimensional consistency issues inside of `calc_slope_functions_with_gradient_model()` are sorted out.